Sources/Datasets/datasets/image/MNIST.swift

```diff
@@ -1,99 +1,108 @@
         switch split {
             case .train:
-                getTrain()
+                Downloader().download(fileAt: URL(string: baseURL + "/train-images-idx3-ubyte.gz")!,
+                                      cacheName: "mnist", fileName: "mnist_train_images") { url, err in
+                                      guard let url = url else {
+                                      fatalError("Data not downloaded.")
+                                      }
+                                      dataURL = url
+                                      semaphore.signal()
+                                    }
+                semaphore.wait()
+                Downloader().download(fileAt: URL(string: baseURL + "/train-labels-idx1-ubyte.gz")!,
+                                      cacheName: "mnist", fileName: "mnist_train_labels") { url, err in
+                                      guard let url = url else {
+                                      fatalError("Data not downloaded.")
+                                      }
+                                      labelURL = url
+                                      semaphore.signal()
+                                    }
+                semaphore.wait()
             case .test:
-                getTest()
-            case .all:
-                getTrain()
-                getTest()
+                Downloader().download(fileAt: URL(string: baseURL + "/t10k-images-idx3-ubyte.gz")!,
+                                      cacheName: "mnist", fileName: "mnist_test_images") { url, err in
+                                      guard let url = url else {
+                                      fatalError("Data not downloaded.")
+                                      }
+                                      dataURL = url
+                                      semaphore.signal()
+                                    }
+                semaphore.wait()
+                Downloader().download(fileAt: URL(string: baseURL + "/t10k-labels-idx1-ubyte.gz")!,
+                                      cacheName: "mnist", fileName: "mnist_test_labels") { url, err in
+                                      guard let url = url else {
+                                      fatalError("Data not downloaded.")
+                                      }
+                                      labelURL = url
+                                      semaphore.signal()
+                                    }
+                semaphore.wait()
             default:
-                break            
+                fatalError("Split does not exist for this dataset.")
         }
     }
 
-    public var info = MNISTInfo
+    private init(batchSize: Int, data: Tensor<UInt8>, labels: Tensor<Int64>) {
+        self.batchSize = batchSize
+        self.data = data
+        self.labels = labels
+    }
+
+    public init(split: S5TFSplit) {
+        self.init(split: split, batchSize: 0)
+    }
```

Comment:
I made a comment on that [here](https://github.com/s5tf-team/datasets/pull/14#discussion_r370927480). Let's resolve them in the same way once we know the solution.